### PR TITLE
Support modal windows (map)

### PR DESCRIPTION
**Pull request by hectoruch**

### CODE REVIEW

File: app/src/components/Map/SupportModal.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+  }
+
+  render() {
+    return (
+      <div className={supportModalStyle['c-support-modal']}>
+        <div className={supportModalStyle['close-modal-button']}>
+          <div className={supportModalStyle.cross}></div>
+        </div>
+        <form>
```
Comment (tiagojsag) [inline app/src/components/Map/SupportModal.jsx]: app/src/components/ContactUs/SupportForm.jsx

**Comment by hectoruch** [comment]
Last change I add the functions to close modal window when user click the background.

File: app/src/components/Map/LayerSupportModal.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+class LayerSupportModal extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      classSelect: '',
+      disabledOption: false
+    };
+    this.handleChange = this.handleChange.bind(this);
```
Comment (andresgnlez) [inline app/src/components/Map/LayerSupportModal.jsx]: are you using this?

File: app/src/components/Map/LayerSupportModal.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        disabledOption: true
+      });
+    }
+  }
+
+  render() {
+    return (
+      <div
+        className={[layerSupportModalStyle['c-layer-back'],
+        layerSupportModalStyle['-support-modal']].join(' ')}
```
Comment (andresgnlez) [inline app/src/components/Map/LayerSupportModal.jsx]: do you know we are using `classnames` for this?

File: app/src/components/Map/SupportModal.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+class SupportModal extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      classSelect: '',
+      disabledOption: false
+    };
+    this.handleChange = this.handleChange.bind(this);
```
Comment (andresgnlez) [inline app/src/components/Map/SupportModal.jsx]: Again, are we using this?

File: app/styles/components/c-button.scss [hunk 5]
```diff
@@ -34,3 +34,18 @@
...
+  display: block;
+  font-size: $font-size-small;
+  letter-spacing: -.1px;
+  margin: 45px auto 0;
+  max-width: 115px;
+  outline: 0;
+  padding: 10px 0;
+  text-transform: uppercase;
+  width: 100%;
+}
```
Comment (andresgnlez) [inline app/styles/components/c-button.scss]: Uhm.. This is not the idea of a css component style. There must be ONE global component with basic and common styles (try to avoid positioning and sizing). Then, use modifiers, inside the component, to get different types of buttons if need it.

Let me know if you need help this. This is important and should be clear, ok? If you need it, we can talk about it.

File: app/styles/components/c-layer-back.scss [hunk 6]
```diff
@@ -0,0 +1,20 @@
+@import '../settings';
+.c-layer-back {//use menu mobile layer
```
Comment (andresgnlez) [inline app/styles/components/c-layer-back.scss]: Try to using a better naming instead of adding comments to say what it is.

File: app/styles/components/c-support-form.scss [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    line-height: 2.21;
+    margin-bottom: 20px;
+    margin-top: 0;
+    text-align: center;
+  }
+
+  form {
+
+    div {
+      float: left;
```
Comment (andresgnlez) [inline app/styles/components/c-support-form.scss]: Why are you using this? Only use `float` when you are very sure you need it, ok? Otherwise this can give you more problems than solutions.

**Comment by andresgnlez** [comment]
You are using `float` and `width: 50%` to set the layout of the modal form. This can be problematic in a nearby future talking about responsiveness. Please use `flex` instead.

File: app/src/components/ContactUs/SupportForm.jsx [hunk 8]
```diff
@@ -49,73 +59,85 @@ class SupportForm extends Component {
...
+          <div className={supportFormStyle['select-container']}>
+            <select
+              id="support_type"
+              onChange={this.handleChange}
+              name="selectCompany"
+              className={supportFormStyle[this.state.classSelect]}
+              required
+            >
+              <option disabled={this.state.disabledOption}>Select an option</option>
+              <option value="Type1">Type 1</option>
```
Comment (tiagojsag) [inline app/src/components/ContactUs/SupportForm.jsx]: Why did you remove the options that were already here?

File: app/src/components/Shared/FooterMini.jsx [hunk 9]
```diff
@@ -34,11 +39,21 @@ class FooterMini extends Component {
         </div>
       );
     }
+    if (this.props.supportModalVisible) {
```
Comment (tiagojsag) [inline app/src/components/Shared/FooterMini.jsx]: why do we need two components here? we already have a modal implemented for login and share, cant we use somtehing like that ?

File: app/src/actions.js [hunk 10]
```diff
@@ -32,6 +32,7 @@ export const UPDATE_FILTERS = 'UPDATE_FILTERS';
 
 // Appearance actions
 export const SET_VISIBLE_MENU = 'SET_VISIBLE_MENU';
+export const SUPPORT_MODAL_OPEN = 'SUPPORT_MODAL_OPEN';
```
Comment (tiagojsag) [inline app/src/actions.js]: we probably don't need an action for this. This is view state, not application state. Nothing outside the map component even knows that there's a support modal window.
